app/Platform/Actions/UpdateAchievementMetricsAction.php

```diff
@@ -62,16 +77,25 @@ public function update(Game $game, Collection $achievements): void
                 + $achievement->points * (($playersHardcoreCalc / $unlocksHardcoreCalc) * $weight)
             );
 
-            $achievement->unlocks_total = $unlocksCount;
-            $achievement->unlocks_hardcore_total = $unlocksHardcoreCount;
-            $achievement->unlock_percentage = $playersTotal ? $unlocksCount / $playersTotal : 0;
-            $achievement->unlock_hardcore_percentage = $playersHardcore ? $unlocksHardcoreCount / $playersHardcore : 0;
+            $bulkUpdates[] = [
+                'ID' => $achievement->ID,
+                'unlocks_total' => $unlocksCount,
+                'unlocks_hardcore_total' => $unlocksHardcoreCount,
+                'unlock_percentage' => $playersTotal ? $unlocksCount / $playersTotal : 0,
+                'unlock_hardcore_percentage' => $playersHardcore ? $unlocksHardcoreCount / $playersHardcore : 0,
+                'TrueRatio' => $pointsWeighted,
+            ];
```

Comment:
Did you not want to have the dirty checks? It's possible that this action could be called in such a way as no updates would occur.

Reply:
Added dirty checks in latest.